mmdet/models/dense_heads/solo_head.py

```diff
@@ -0,0 +1,479 @@
+import os
+from scipy import ndimage
+import mmcv
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from mmdet.core import bbox2result, multi_apply
+from mmcv.cnn import ConvModule, build_upsample_layer, bias_init_with_prob
+from .base_dense_head import BaseDenseHead
+from mmdet.core import matrix_nms
+from mmdet.models.builder import HEADS, build_loss
+
```

Comment:
consider making SOLOHead a base class of DecoupledSOLOHead to reduce the code of duplicated segm2results.